While recovering services, don't skip if one of the service recovery fails.

chowmean opened the pull request:
While recovering a service doesn't return an error instead try to recover the remaining service. 
Fixes: [#18371](https://github.com/cilium/cilium/issues/18371)

```release-note
In the case of recovering the services, cilium will not fail directly on the first service recovery error but will try to recover other services. 
```

kkourt reviewed (changes requested):
Hi, thanks for the patch!

Can you please add some additional details in the commit message? Rewording or even copy-pasting the text from the issue would be fine. The goal here would be to have a brief description of what the problem that this patch tries to solve is and how it manages it.

Thanks again!

kkourt commented on pkg/service/service.go:
```diff
@@ -851,8 +851,10 @@ func (s *Service) restoreBackendsLocked() error {
+			log.WithError(err).WithField(logfields.BackendID, b.ID).
+				Warnf("Unable to restore backend ID %d for %q: %s",
+					b.ID, b.L3n4Addr, err)
+			continue
```
How about putting all arguments as fields and removing them from the message?

```
log.WithFields(logrus.Fields{
                logfields.BackendID: b.ID,
		logfields.L3n4Addr: b.L3n4Addr,
	}).Warn("Unable to restore backend")
```

chowmean replied:
done

kkourt commented on pkg/service/service.go:
```diff
@@ -935,7 +937,8 @@ func (s *Service) restoreServicesLocked() error {
 			}
 			if err := s.lbmap.UpsertMaglevLookupTable(uint16(newSVC.frontend.ID), backends, ipv6); err != nil {
-				return err
+				scopedLog.WithError(err).Warning("Unable to Inserts into the Maglev BPF map.")
```
How about:

> scopedLog.WithError(err).Warning("Unable to upsert into the Maglev BPF map.")

?

chowmean replied:
Changed it to this

kkourt reviewed (approved):
Thanks! some more nits from my side, but otherwise LGTM.

brb reviewed (changes requested):
My comments are the same as what @kkourt said. Also, it would be good to log a count for how many backends failed to be restored (same it is done for the services).

brb commented on pkg/service/service.go:
```diff
@@ -851,14 +852,23 @@ func (s *Service) restoreBackendsLocked() error {
+			log.WithFields(logrus.Fields{
+				logfields.BackendID: b.ID,
+				logfields.L3n4Addr:  b.L3n4Addr,
+			}).Warn("Unable to restore backend")
```
The error needs to be logged.

chowmean replied:
Made the change. Please have a look.

brb commented on pkg/service/service.go:
```diff
@@ -851,14 +852,23 @@ func (s *Service) restoreBackendsLocked() error {
+			log.WithFields(logrus.Fields{
+				logfields.BackendID: b.ID,
+				logfields.L3n4Addr:  b.L3n4Addr,
+			}).Error("Unable to restore backend: %s", err)
```
Please use `WithError()` instead.

chowmean replied:
made the changes.

brb commented on pkg/service/service.go:
```diff
@@ -851,14 +852,23 @@ func (s *Service) restoreBackendsLocked() error {
 	}
 
+	log.WithFields(logrus.Fields{
+		"restored": restored,
```
Please use `pkg/logging/logfields/` for `"restored"` and `"failed"`. If there is no suitable logfield, then you can introduce it.

chowmean replied:
Added the new variable please take look.

brb reviewed (changes requested):
Thanks! One more nit before we can merge it - could you squash both commits into one?

brb commented on pkg/service/service.go:
```diff
@@ -851,14 +852,23 @@ func (s *Service) restoreBackendsLocked() error {
+			log.WithError(err).WithFields(logrus.Fields{
+				logfields.BackendID: b.ID,
+				logfields.L3n4Addr:  b.L3n4Addr,
+			}).Error("Unable to restore backend")
```
The last nit: let's use `Warning` instead of `Error` here.

chowmean replied:
Made the change

chowmean committed:
In service recovery, don't skip if one of the service recovery fails

Reduce number of connections interrupted due to svc/backend
ID changes if restoration fails

Fixed by not return error instead logging it and continue.

Logging number of failures and success in backend restoration
with new variables

Signed-off-by: Gaurav Yadav <gaurav.dev.iiitm@gmail.com>

brb reviewed (approved):
Thanks!
